usr/lib/common/pkey_utils.c

```diff
@@ -1196,31 +1196,55 @@ CK_RV pkey_ec_sign(STDLL_TokData_t *tokdata, SESSION *session,
     curve_type = get_cpacf_curve_type(privkey->template);
     switch (curve_type) {
     case curve_p256:
+        if (pkey_attr->ulValueLen != sizeof(param.P256.priv) + sizeof(param.P256.vp)) {
+            TRACE_ERROR("Protected key has an invalid length of %ld bytes.\n",
+                        pkey_attr->ulValueLen);
+            ret = CKR_ATTRIBUTE_VALUE_INVALID;
+            goto done;
+        }
         if (hash_len > 32)
             hash_len = 32;
-        off = 32 - hash_len;
+        off = sizeof(param.P256.hash) - hash_len;
         memcpy(param.P256.hash + off, hash, hash_len);
-        memcpy(param.P256.priv, pkey_attr->pValue, 32);
-        memcpy(param.P256.vp, (char *)pkey_attr->pValue + 32, 32);
+        memcpy(param.P256.priv, pkey_attr->pValue,
+               pkey_attr->ulValueLen - sizeof(param.P256.vp));
+        memcpy(param.P256.vp, (char *)pkey_attr->pValue + sizeof(param.P256.priv),
+               sizeof(param.P256.vp));
         *sig_len = 2 * 32;
```

Comment:
`*sig_len = 2 * sizeof(param.P256.sig_r)` here ? 

same for the other curves below.

Reply:
ok, fixed and re-pushed.